File: packages/ui-flutter/lib/src/tab_bar.dart
import 'package:flutter/widgets.dart';

import 'theme.dart';

/// Mirrors `packages/ui/src/components/tab-bar.tsx`: a mobile bottom
/// navigation bar — a fixed row of icon + label destinations, active item
/// driven by `isActive`, optional `badge`.
class KinetixTabBar extends StatelessWidget {
  const KinetixTabBar({super.key, required this.children});

  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return Container(
      decoration: BoxDecoration(
        color: c.background,
        border: Border(top: BorderSide(color: c.border, width: 1)),
      ),
      child: Row(children: [for (final item in children) Expanded(child: item)]),
    );
  }
}

class KinetixTabBarItem extends StatelessWidget {
  const KinetixTabBarItem({
    super.key,
    required this.label,
    required this.icon,
    required this.onTap,
    this.isActive = false,
    this.badge,
  });

  final String label;
  final Widget icon;
  final VoidCallback onTap;
  final bool isActive;
  final String? badge;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    final Color color = isActive ? c.primary : c.mutedForeground;

    return GestureDetector(
      onTap: onTap,
      behavior: HitTestBehavior.opaque,
      child: Padding(
        padding: const EdgeInsets.symmetric(vertical: 8),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Stack(
              clipBehavior: Clip.none,
              children: [
                IconTheme.merge(
                  data: IconThemeData(color: color, size: 20),
                  child: icon,
                ),
                if (badge != null)
                  Positioned(
                    right: -8,
                    top: -6,
                    child: Container(
                      constraints: const BoxConstraints(minWidth: 16, minHeight: 16),
                      padding: const EdgeInsets.symmetric(horizontal: 4),
                      alignment: Alignment.center,
                      decoration: BoxDecoration(
                        color: c.destructive,
                        borderRadius: BorderRadius.circular(9999),
                      ),
                      child: Text(
                        badge!,
                        style: TextStyle(
                          fontSize: 10,
                          fontWeight: FontWeight.w500,
                          color: c.destructiveForeground,
                        ),
                      ),
                    ),
                  ),
              ],
            ),
            const SizedBox(height: 4),
            Text(label, style: TextStyle(fontSize: 11, fontWeight: FontWeight.w500, color: color)),
          ],
        ),
      ),
    );
  }
}

File: packages/ui-swiftui/Sources/KinetixUI/TabBar.swift
//
// TabBar.swift — KinetixTabBar / KinetixTabBarItem.
//
// Mirrors packages/ui/src/components/tab-bar.tsx: a mobile bottom
// navigation bar — a fixed row of icon + label destinations, active item
// driven by `isActive`, optional `badge`.
//

import SwiftUI

public struct KinetixTabBar<Content: View>: View {
    @Environment(\.kinetixColors) private var colors
    private let content: Content

    public init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    public var body: some View {
        HStack(spacing: 0) { content }
            .frame(maxWidth: .infinity)
            .background(colors.background)
            .overlay(alignment: .top) {
                Rectangle().fill(colors.border).frame(height: 1) // border-t
            }
    }
}

public struct KinetixTabBarItem<Icon: View>: View {
    @Environment(\.kinetixColors) private var colors

    private let label: String
    private let isActive: Bool
    private let badge: String?
    private let icon: Icon
    private let action: () -> Void

    public init(
        label: String,
        isActive: Bool = false,
        badge: String? = nil,
        action: @escaping () -> Void,
        @ViewBuilder icon: () -> Icon
    ) {
        self.label = label
        self.isActive = isActive
        self.badge = badge
        self.action = action
        self.icon = icon()
    }

    public var body: some View {
        Button(action: action) {
            VStack(spacing: 4) {
                icon
                    .font(.system(size: 20))
                    .overlay(alignment: .topTrailing) {
                        if let badge {
                            Text(badge)
                                .font(.system(size: 10, weight: .medium))
                                .foregroundStyle(colors.destructiveForeground)
                                .padding(.horizontal, 4)
                                .frame(minWidth: 16, minHeight: 16)
                                .background(colors.destructive, in: Capsule())
                                .offset(x: 8, y: -6)
                        }
                    }
                Text(label)
                    .font(.kinetixLabelSm)
            }
            .foregroundStyle(isActive ? colors.primary : colors.mutedForeground)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 8)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityAddTraits(isActive ? [.isSelected] : [])
    }
}
